src/natcap/invest/urban_flood_risk_mitigation.py

```diff
@@ -322,107 +313,245 @@ def execute(args):
+        # Determine flood_volume over the watershed
+        flood_volume_in_aoi_task = task_graph.add_task(
+            func=pygeoprocessing.zonal_statistics,
+            args=(
+                (runoff_retention_raster_path, 1),
+                reprojected_aoi_path),
+            dependent_task_list=[flood_vol_task],
+            task_name='zonal_statistics over the flood_volume raster')
+        flood_volume_stats = flood_volume_in_aoi_task.get()
+        damage_per_aoi_stats = damage_to_infrastructure_in_aoi_task.get()
+
+    summary_vector_path = os.path.join(
+        args['workspace_dir'], 'flood_risk_service%s.shp' % file_suffix)
+    write_summary_vector_stats = task_graph.add_task(
+        func=_write_summary_vector,
+        args=(reprojected_aoi_path,
+              summary_vector_path),
+        kwargs={
+            'runoff_ret_stats': runoff_retention_stats_task.get(),
+            'runoff_ret_vol_stats': runoff_retention_volume_stats_task.get(),
+            'damage_per_aoi_stats': damage_per_aoi_stats,
+            'flood_volume_stats': flood_volume_stats,
+        },
+        target_path_list=[summary_vector_path],
+        task_name='write summary stats to flood_risk_service.shp',
+        dependent_task_list=summary_tasks)
 
     task_graph.close()
     task_graph.join()
 
 
-def _pickle_zonal_stats(
-        base_vector_path, base_raster_path, target_pickle_path):
-    """Calculate Zonal Stats for a vector/raster pair and pickle result.
+def _write_summary_vector(
+        source_aoi_vector_path, target_vector_path, runoff_ret_stats=None,
+        runoff_ret_vol_stats=None, damage_per_aoi_stats=None,
+        flood_volume_stats=None):
 
-    Parameters:
-        base_vector_path (str): path to vector file
-        base_raster_path (str): path to raster file to aggregate over.
-        target_pickle_path (str): path to desired target pickle file that will
-            be a pickle of the pygeoprocessing.zonal_stats function.
+    source_aoi_vector = gdal.OpenEx(source_aoi_vector_path, gdal.OF_VECTOR)
+    source_aoi_layer = source_aoi_vector.GetLayer()
+    source_geom_type = source_aoi_layer.GetGeomType()
+    source_srs_wkt = pygeoprocessing.get_vector_info(
+        source_aoi_vector_path)['projection']
+    source_srs = osr.SpatialReference()
+    source_srs.ImportFromWkt(source_srs_wkt)
 
-    Returns:
-        None.
+    esri_driver = gdal.GetDriverByName('ESRI Shapefile')
+    target_watershed_vector = esri_driver.Create(
+        target_vector_path, 0, 0, 0, gdal.GDT_Unknown)
+    layer_name = str(os.path.splitext(os.path.basename(
+        target_vector_path))[0])
+    LOGGER.debug("creating layer %s", layer_name)
+    target_watershed_layer = target_watershed_vector.CreateLayer(
+        str(layer_name), source_srs, source_geom_type)
+
+    target_fields = ['rnf_rt_idx', 'rnf_rt_m3']
+    if not damage_per_aoi_stats:
+        damage_per_aoi_stats = {}
+    else:
+        target_fields += ['aff_bld', 'serv_bld']

```

Comment:
It is indeed, and has been patched in 91bbc64d.  That commit also improves the tests so that this oversight would have been caught had they been in place when I was refactoring the model.